operator/pkg/lbipam/lbipam.go

```diff
@@ -0,0 +1,1791 @@
+		return false
+	}
+
+	return lastCondition.Status == meta_v1.ConditionTrue
+}
+
+// settleConflicts check if there exist any un-resolved conflicts between the ranges of IP pools and resolve them.
+// secondly, it checks if any ranges that are marked as conflicting have been resolved.
+// Any found conflicts are reflected in the IP Pool's status.
+func (ipam *LBIPAM) settleConflicts() error {
+	ipam.logger.Debug("Settling pool conflicts")
+
+	// Mark any pools that conflict as conflicting
+	for _, poolOuter := range ipam.pools {
+		if isPoolConflicting(poolOuter) {
+			continue
+		}
+
+		outerRanges, _ := ipam.rangesStore.GetRangesForPool(poolOuter.GetName())
+
+		if conflicting, rangeA, rangeB := areRangesInternallyConflicting(outerRanges); conflicting {
+			err := ipam.markPoolConflicting(poolOuter, poolOuter, rangeA, rangeB)
+			if err != nil {
+				return fmt.Errorf("markPoolConflicting: %w", err)
+			}
+			continue
+		}
+
+		for _, poolInner := range ipam.pools {
+			if poolOuter.GetName() == poolInner.GetName() {
+				continue
+			}
+
+			if isPoolConflicting(poolInner) {
+				continue
+			}
+
+			innerRanges, _ := ipam.rangesStore.GetRangesForPool(poolInner.GetName())
+			if conflicting, outerRange, innerRange := areRangesConflicting(outerRanges, innerRanges); conflicting {
+				// If two pools are conflicting, disable/mark the newest pool
+
+				if poolOuter.CreationTimestamp.Before(&poolInner.CreationTimestamp) {
+					err := ipam.markPoolConflicting(poolInner, poolOuter, innerRange, outerRange)
+					if err != nil {
+						return fmt.Errorf("markPoolConflicting: %w", err)
+					}
+					break
+				}
+
+				err := ipam.markPoolConflicting(poolOuter, poolInner, outerRange, innerRange)
+				if err != nil {
+					return fmt.Errorf("markPoolConflicting: %w", err)
+				}
+				break
+			}
+		}
+	}
+
+	// un-mark pools that no longer conflict
+	for _, poolOuter := range ipam.pools {
+		if !isPoolConflicting(poolOuter) {
+			continue
+		}
+
+		outerRanges, _ := ipam.rangesStore.GetRangesForPool(poolOuter.GetName())
+
+		// If the pool is still internally conflicting, don't un-mark
+		if conflicting, _, _ := areRangesInternallyConflicting(outerRanges); conflicting {
+			continue
+		}
+
+		poolConflict := false
+		for _, poolInner := range ipam.pools {
+			if poolOuter.GetName() == poolInner.GetName() {
+				continue
+			}
+
+			innerRanges, _ := ipam.rangesStore.GetRangesForPool(poolInner.GetName())
+			if conflicting, _, _ := areRangesConflicting(outerRanges, innerRanges); conflicting {
+				poolConflict = true
+				break
+			}
+		}
+
+		// The outer pool, which is marked conflicting no longer conflicts
+		if !poolConflict {
+			err := ipam.unmarkPool(poolOuter)
+			if err != nil {
+				return fmt.Errorf("unmarkPool: %w", err)
+			}
+		}
+	}
+
+	return nil
+}
+
+// markPoolConflicting marks the targetPool as "Conflicting" in its status and disables all of its ranges internally.
+func (ipam *LBIPAM) markPoolConflicting(
```

Comment:
nit: as a general comment, it looks like LBIPAM has a large method set, and some of those methods could actually become part of the objects they receive as parameters. 

if it makes sense, consider inverting the responsibility to the structures themselves, not LBIPAM manipulating them.

Reply:
This is inline with feedback from @joamaki as well. Writing the logic this way is easy since all of the methods have all of the data available they might need.

In a future PR, we are going to attempt to split up this logic into separate cells so each part can request the objects they need and let hive sort out the dependencies and init order.